### PR TITLE
feat: integrated wagmi and wagmiConfig, improved data flow and state management

### PR DESCRIPTION
**Pull request by sstefdev**
Fixes https://github.com/RequestNetwork/web-components/issues/31

## Release Notes

- **New Features**
	- Enhanced account management in the invoice creation and dashboard components using the Wagmi library.
	- New `wagmiConfig` property added for improved wallet integration.
	- Introduced `getEthersSigner` function for better signer management.
	- Simplified usage instructions in the documentation for integrating components into React applications.

- **Bug Fixes**
	- Updated logic for handling Ethereum addresses to ensure proper formatting and assignment.

- **Documentation**
	- Updated example usages in documentation to reflect changes in component properties.

- **Chores**
	- Added new dependencies to improve functionality and maintainability of the invoice form and dashboard packages.
	- Default parameter added to `initializeCurrencyManager` for better usability.

### CODE REVIEW

File: packages/invoice-dashboard/src/lib/view-requests.svelte [hunk 1]
```diff
@@ -76,7 +77,26 @@
...
+      watchAccount(wagmiConfig, {
+        onChange(data) {
+          if (data.address !== account?.address) {
+            account = data;
+            getRequests();
+          }
+        },
+      });
+    }
+  }
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply packages/invoice-dashboard/src/lib/view-requests.svelte]: It does require wagmi config...

File: packages/create-invoice-form/src/lib/create-invoice-form.svelte [hunk 2]
```diff
@@ -77,7 +79,13 @@
   };
 
   $: {
-    formData.creatorId = signer;
+    if (wagmiConfig) {
+      account = getAccount(wagmiConfig);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply packages/create-invoice-form/src/lib/create-invoice-form.svelte:84]: You're hallucinating. The getAccount function *does* take a parameter. Resolving as no problem :call_me_hand:

File: packages/invoice-dashboard/src/lib/view-requests.svelte [hunk 3]
```diff
@@ -76,7 +80,45 @@
...
+            previousWalletAddress = undefined;
+            previousNetwork = undefined;
+          }
+        }
+      },
+    });
+  });
+
+  onDestroy(() => {
+    if (unwatchAccount) unwatchAccount();
```
Comment (MantisClone) [inline packages/invoice-dashboard/src/lib/view-requests.svelte:117]: This conditional makes `unwatchAccount` seem like a boolean. Recommend adjusting to make it clear that we're checking for existence of a function.

```suggestion
    if (typeof unwatchAccount === 'function') unwatchAccount();
```

**Review by MantisClone — APPROVED** [review]
LGTM!

In the future, I recommend responding to CodeRabbit comments to explain why each comment is resolved. 

* Resolved because you fixed it
* Resolved because you disagree with the comment and there's actually no Problem
* Resolved because you agree with the comment but you won't fix it now.

**Pull request merged**